### PR TITLE
Feature/pd 380 ensure import creates kp is

### PR DESCRIPTION
**Pull request by gortonz**
Added PD-380 changes

### CODE REVIEW

File: lib/tasks/import.rake [hunk 1]
```diff
@@ -13,9 +13,39 @@ namespace :import do
...
+      data['datasets'].each do |ds|
+        if ds['data'] == nil && isKPI(ds['id'], kpiComponents)
+            d = Hash.new
+            d['label'] = Time.now.strftime('%Y-%m')
+            d['value'] =  0
+            ds['data'] = [1]
+            ds['data'][0] = d
+        end
+      end
+      # puts "#{data}"
```
Comment (elisechant) [inline lib/tasks/import.rake]: remove this line

File: lib/tasks/import.rake [hunk 2]
```diff
@@ -13,9 +13,39 @@ namespace :import do
...
+
       file = File.read("lib/data/#{name}-definition.json")
       definition = JSON.parse(file)
```
Comment (elisechant) [inline lib/tasks/import.rake]: any reason why we need this twice?

File: lib/tasks/import.rake [hunk 3]
```diff
@@ -13,9 +13,39 @@ namespace :import do
...
+        # this layout is not in a correct form
+        layoutKPI=[["kpis"], kpiComponents]
+        for i in 0..layout.length-1
+          if (layout[i] <=> ["kpis"]) != 0 && (layout[i]<=>kpiComponents) != 0 then
+            layoutKPI.push(layout[i])
+          end
+        end
+        definition['layout']=layoutKPI
+      end
+      # puts "#{definition['layout']}"
```
Comment (elisechant) [inline lib/tasks/import.rake]: can remove this

File: lib/tasks/import.rake [hunk 4]
```diff
@@ -33,7 +63,6 @@ namespace :import do
         # puts dataset['id']
```
Comment (elisechant) [inline lib/tasks/import.rake]: can remove this

File: app/views/dashboards/_overview_widget.html.erb [hunk 5]
```diff
@@ -5,5 +5,27 @@
     </div>
     <div class="widget__inner" data-data="" aria-hidden="true">
     </div>
+    <div class="widget__data">
+      <table class="data-table table">
+        <%  widget.datasets.each do |dataset| %>
+          <caption>"<%= widget.name %>"</caption>
```
Comment (elisechant) [inline app/views/dashboards/_overview_widget.html.erb]: we don't need the " "s inside caption

File: lib/tasks/import.rake [hunk 6]
```diff
@@ -13,9 +13,39 @@ namespace :import do
...
       file = File.read("lib/data/#{name}-definition.json")
       definition = JSON.parse(file)
 
+      # check the definition layout section and make sure the KPIs are specified in it
+      layout = definition['layout']
+
+      if (layout[0]<=>["kpis"])==0 && (layout[1]<=>kpiComponents)==0 then
+        # it is the layout what we need
+      else
+        # this layout is not in a correct form
```
Comment (elisechant) [inline lib/tasks/import.rake]: what does this mean?

File: lib/tasks/import.rake (hunk 1 above)
Comment (tobyhede) [inline lib/tasks/import.rake]: We should just leave the KPI datasets as null. 0 is not the same as no data

File: lib/tasks/import.rake [hunk 7]
```diff
@@ -13,9 +13,39 @@ namespace :import do
       file = File.read("lib/data/#{name}-data.json")
       data = JSON.parse(file)
 
+      # check the data, if KPIs data are not set, we will add some default values in them
+      kpiComponents = %w(user-satisfaction cost-per-transaction digital-take-up completion-rate)
```
Comment (tobyhede) [inline lib/tasks/import.rake]: Ruby convention is snake_case (kpi_components)

File: lib/tasks/import.rake [hunk 8]
```diff
@@ -69,4 +98,15 @@ namespace :import do
     end
 
   end
+
+  def isKPI(kpiId, kpiComponents)
```
Comment (tobyhede) [inline lib/tasks/import.rake]: So. Ruby is semi-functional and excellent. 

`kpiComponents.include?(kpi_id)`

**Comment by tobyhede** [comment]
Closing this as I think we can now just fix in the data.

**Pull request closed without merging**